src/protocol/kademlia/impl/peer_routing_table_impl.cpp

```diff
@@ -119,22 +118,40 @@ namespace libp2p::protocol::kademlia {
 
       // the structure of the table has changed, so let's recheck if the peer
       // now has a dedicated bucket.
-      auto bid = cpl;
-      if (bid >= buckets_.size()) {
-        bid = buckets_.size() - 1;
+      auto resizedBucketId = getBucketId(buckets_, cpl);
+      auto &resizedBucket = buckets_.at(resizedBucketId);
+      if (resizedBucket.size() < config_.maxBucketSize) {
+        resizedBucket.emplace_front(pid, is_permanent);
+        bus_->getChannel<events::PeerAddedChannel>().publish(pid);
+        return true;
       }
+      auto replaceablePeerIt =
+          std::find_if(resizedBucket.begin(), resizedBucket.end(),
```

Comment:
Perhaps is better to use reversive search. Eldest peer ids are expected it tail.
```suggestion
          std::find_if(resizedBucket.rbegin(), resizedBucket.rend(),
```